tests/broker-engine/services-and-bulk-stmt.robot

```diff
@@ -286,67 +286,117 @@ RLCode
+    Should Be True    ${result}    msg="lua logs produced"
 
     # Define the new content to take place of the first one
-    ${new_content}    Catenate
+    ${new_content}=    Catenate
     ...    function init(params)
-    ...    broker_log:set_parameters('/tmp/titi.log', 2)
+    ...        broker_log:set_parameters('/tmp/titi.log', 2)
     ...    end
     ...
     ...    function write(d)
-    ...    broker_log:info(0, "titi")
-    ...    return true
+    ...        broker_log:info(0, "titi")
+    ...        return true
     ...    end
 
+
     # Create the second LUA script file
     Create File    /tmp/toto.lua    ${new_content}
-    ${start}    Get Current Date
+    ${start}=    Get Current Date
 
     Reload Broker
 
-    ${content}    Create List    lua: initializing the Lua virtual machine
-    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
-    Should Be True    ${result}    lua file not initialized
+    ${content}=    Create List    lua: initializing the Lua virtual machine
+    ${result}=    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
+    Should Be True    ${result}    msg=lua file not initialized
+
+
+    Stop Engine
+    Kindly Stop Broker
+
+metric_mapping
+    [Documentation]    check if metric name is exist using lua conf
+    [Tags]    broker    engine    bbdo    unified_sql    metric
+    Clear Commands Status
+    Clear Retention
+
+    Remove File    /tmp/test4.log
+    Config Engine    ${1}    ${1}    ${10}
+    Config Broker    central
+    Config Broker    module
+    Broker Config Add Item    central    bbdo_version    3.0.1
+    Broker Config Add Item    module0    bbdo_version    3.0.1
+    Broker Config Log    central    lua    debug
+    Broker Config Log    module0    neb    debug
+    Config Broker Sql Output    central    unified_sql
 
+    Broker Config Add Lua Output    central    test-metric    ${SCRIPTS}test-metric.lua
+
+    ${start}    Get Current Date
+
+    Start Broker
+    Start Engine
+
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    A message telling check_for_external_commands() should be available.
+
+    # Let's wait for one "INSERT INTO data_bin" to appear in stats.
+    FOR    ${i}    IN RANGE    ${10}
+        Process Service Check result with metrics    host_1    service_${i+1}    1    warning${i}    20
+    END
+
+    Wait Until Created    /tmp/test4.log    1m
+
+    ${metric_name_found}    Set Variable    False
+    FOR    ${i}    IN RANGE    10
+        ${grep_res}    Grep File
+        ...    /tmp/test4.log
+        ...    "\"name\": \"metric_${i}\""
+        Sleep    1s
+        Run Keyword If    "${grep_res}" != ""    Set Variable    ${metric_name_found}    ${True}
+        Exit For Loop If    "${grep_res}" != ""
+    END
```

Comment:
Instead of that loop, you can use the `Wait Until File Created` and then check its content.

Reply:
i removed the loop can u check if that new modification is accepted